### PR TITLE
Add CFD and Commodity support for Interactive Brokers

### PR DESCRIPTION
**Pull request by DracheShiki**
# Pull Request

add CFD and Commodity support for Interactivebrokers;
fix the strict_symbology for parsing Instrument/contract of Interactivebrokers

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How has this change been tested?

Tested with "XAUUSD.IBCMDTY", "XAUUSD.IBCFD", "IBUS30.IBCFD", "TSLA.IBCFD", "EUD/USD.IBCFD" by live trading and historical data downloading. 
a doc that "IBCFD" and "IBCMDTY" used as venue (mapping to "SMART") might be updated.
Although it works with no bugs as far as i tested it, the cfd.pyx and commodity.pyx may need to be refined.

Thanks @rsmb7z for advice.

### CODE REVIEW

**Comment by cjdsellers** [comment]
Hi @DracheShiki

Thanks so much for this awesome contribution :pray:.

Looks like we just have pre-commit failing so far. You can run pre-commit with:
```
make pre-commit
```
or
```
pre-commit run --all-files
```

You can also run the following to install the git hook to ensure it always runs on local commits too:
```
pre-commit install
```

It looks like some auto-formatting from `black` will occur, then you can push another commit.

File: nautilus_trader/adapters/interactive_brokers/parsing/instruments.py [hunk 1]
```diff
@@ -74,8 +76,16 @@
     "NYBOT",  # US
     "SNFE",  # AU
 ]
+VENUES_CFD = [
```
Comment (rsmb7z) [inline nautilus_trader/adapters/interactive_brokers/parsing/instruments.py:79]: @DracheShiki all of these are translating to `SMART`? If so just use one i.e `IBCFD` to keep things simple.

**Comment by rsmb7z** [comment]
LGTM.

**Review by rsmb7z — APPROVED** [review]

**Comment by DracheShiki** [comment]
> Hi @DracheShiki
> 
> Thanks so much for this awesome contribution 🙏.
> 
> Looks like we just have pre-commit failing so far. You can run pre-commit with:
> 
> ```
> make pre-commit
> ```
> 
> or
> 
> ```
> pre-commit run --all-files
> ```
> 
> You can also run the following to install the git hook to ensure it always runs on local commits too:
> 
> ```
> pre-commit install
> ```
> 
> It looks like some auto-formatting from `black` will occur, then you can push another commit.

oh i just pushed the old one back-up without pre-commit.
I will push one with it. But for the function "ib_contract_to_instrument_id_simplified_symbology(contract: IBContract)" i can't reduce the complex...

File: nautilus_trader/adapters/interactive_brokers/parsing/instruments.py (hunk 1 above)
Comment (rsmb7z) [reply nautilus_trader/adapters/interactive_brokers/parsing/instruments.py:79]: Understood, lets leave it the way you did.

**Comment by rsmb7z** [comment]
> I will push one with it. But for the function "ib_contract_to_instrument_id_simplified_symbology(contract: IBContract)" i can't reduce the complex...

For `ib_contract_to_instrument_id_simplified_symbology` and `instrument_id_to_ib_contract_simplified_symbology` at the end of line add comment `# noqa: C901`.

**Comment by DracheShiki** [comment]
> > I will push one with it. But for the function "ib_contract_to_instrument_id_simplified_symbology(contract: IBContract)" i can't reduce the complex...
> 
> For `ib_contract_to_instrument_id_simplified_symbology` and `instrument_id_to_ib_contract_simplified_symbology` at the end of line add comment `# noqa: C901`.

well, i just get confused for a while that everytime i added "#noqa: C901" then i excuted "pre-commit run --all-files", the "#noqa: C901" would get deleted by precommit...Anyway i pushed a version with "#noqa: C901" and without excuting "pre-commit run --all-files" again xD

**Comment by DracheShiki** [comment]
> > well, i just get confused for a while that everytime i added "#noqa: C901" then i excuted "pre-commit run --all-files", the "#noqa: C901" would get deleted by precommit...Anyway i pushed a version with "#noqa: C901" and without excuting "pre-commit run --all-files" again xD
> 
> The reason it gets deleted is because it expects on the first (def) line, even if its multi-line def.

Oh thanks…That is also out of my cognition.

**Pull request merged**